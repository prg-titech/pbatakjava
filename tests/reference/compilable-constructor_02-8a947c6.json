{
    "basename": "compilable-constructor_02-8a947c6",
    "cmd": "{infile}",
    "infile": "tests/constructor_02/build.sh",
    "infile_hash": "1f66bf060e2319d8b217a108f34df5c0d2031d4388a8a2bb2828a7ec",
    "stdout": "compilable-constructor_02-8a947c6.stdout",
    "stdout_hash": "adb23c4645287e26411bdb71bf55a1a8ae68d81101be2e64dcc6f1ac",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}