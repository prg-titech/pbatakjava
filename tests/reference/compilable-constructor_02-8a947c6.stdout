
public interface Point {
  public int get__x();

  public int set__x(int toAssign);

  public int get__y();

  public int set__y(int toAssign);

  public int get__z();

  public int set__z(int toAssign);
}

public interface Point_Factory {
  public Point make(int x, int y);

  public Point make(int x, int y, int z);
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }

  public int get__z() {
    return 0;
  }

  public int set__z(int toAssign) {
    return 0;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }

  public Point make(int x, int y, int z) {
    return null;
  }
}

public class Point_v2 implements Point {
  public int x;

  public int y;

  public int z;

  public Point_v2(int x, int y, int z) {
    this.x = x;
    this.y = y;
    this.z = z;
  }

  public Point_v2(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }

  public int get__z() {
    return this.z;
  }

  public int set__z(int toAssign) {
    return this.z = toAssign;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y, int z) {
    return new Point_v2(x, y, z);
  }

  public Point make(int x, int y) {
    return new Point_v2(x, y);
  }
}

public class Point_v3 implements Point {
  public int x;

  public int y;

  public int z;

  public Point_v3(int x, int y, int z) {
    this.x = x;
    this.y = y;
    this.z = z;
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }

  public int get__z() {
    return this.z;
  }

  public int set__z(int toAssign) {
    return this.z = toAssign;
  }
}

public class Point_v3_Factory implements Point_Factory {
  public Point make(int x, int y, int z) {
    return new Point_v3(x, y, z);
  }

  public Point make(int x, int y) {
    return null;
  }
}
